### PR TITLE
🔒 Execution Lock Mechanism

### PR DESCRIPTION
**Pull request by JaredBorders**
Add lock mechanism to account execution. The lock can be turned on/off by the new `Settings` owner.

## Description
* Create `Settings.sol` (long time no see 😅)
* Update `Account.sol` constructor to include `Settings.sol` address
* Add interface to `Account.sol` (`ISettings.sol`)
* Add check to `Account.sol`'s `execute()` function to ensure system not locked
* Add check to `executeConditionalOrder()` function to ensure system not locked
* Add new custom error to `IAccount.sol` thrown when the system is locked
* Update deploy script (add settings and a constants file to clean it up a bit)
* Add tests for `Settings.sol`
* Add new class of tests for implementation upgrades

## Related issue(s)
N/A

## Motivation and Context
https://github.com/Kwenta/kwenta-state-log/blob/master/kips/drafts/kip-73.md

### CODE REVIEW

File: test/integration/order.behavior.t.sol [hunk 1]
```diff
@@ -76,7 +78,8 @@ contract OrderBehaviorTest is Test, ConsolidatedEvents {
             futuresMarketManager, 
             address(systemStatus), 
             GELATO, 
-            OPS
+            OPS,
+            address(0)
```
Comment (tommyrharper) [inline test/integration/order.behavior.t.sol:82]: **Question:** Why is `address(settings)` not passed in here?

File: test/integration/upgrade.behavior.t.sol
Comment (tommyrharper) [inline test/integration/upgrade.behavior.t.sol:1]: **Thought:** Might be nice to have an `Upgrade.s.sol` script for this.

**Review by tommyrharper — APPROVED** [review]
LGTM 👍 🚀

File: test/integration/order.behavior.t.sol (hunk 1 above)
Comment (JaredBorders) [reply test/integration/order.behavior.t.sol]: Updated. Not intentional

File: src/interfaces/IAccount.sol [hunk 2]
```diff
@@ -6,6 +6,7 @@ import {IFactory} from "./IFactory.sol";
 import {IFuturesMarketManager} from "@synthetix/IFuturesMarketManager.sol";
 import {IPerpsV2MarketConsolidated} from
     "@synthetix/IPerpsV2MarketConsolidated.sol";
+import {ISettings} from "./ISettings.sol";
```
[the remark this answers is not in the document]
Comment (JaredBorders) [reply src/interfaces/IAccount.sol:9]: discussed irl

File: test/integration/upgrade.behavior.t.sol
Comment (JaredBorders) [reply test/integration/upgrade.behavior.t.sol:1]: All I can really do is deploy a new implementation (which already exists). Multisig has to actually call the factory to update

**Pull request merged**